src/com/visualization/piechart/piechart.js

```diff
@@ -0,0 +1,69 @@
+import {h, Component} 				from 'preact/preact';
+
+import PieSegment					from 'com/visualization/piechart/piesegment';
+
+export default class PieChart extends Component {
+
+	constructor( props ) {
+		super(props);
+
+    }
+
+    // Converts an array of number to an array of percentages
+    convertToPercentage ( values ) {
+        // calculate total
+        let total = 0;
+        values.forEach( ( v ) => ( total += v ) );
+
+        // scale values so there sum is 100%
+        let percentages = values.map( ( v ) => ( (100 * v) / total ) );
+
+        return percentages;
+    }
+
+    render( props ) {
+
+        if ( !(props && props['lables'] && props['values'])) {
+            console.warn('PieChart was created with invalid props', props);
+            return <div> No Data! </div>;
+        }
+
+        let lables = props.lables;
+        let values = props.values;
+
+        let percentages = this.convertToPercentage(values);
+
+        let Segments = [];
+        let Ledgend = [];
+
+        let offset = 0;
+        for (var i = 0; i < percentages.length; i++) {
+
+            let color = 1 + ( i % 6 );
+            let ledgendclass = "shape-circle ledgend_color_"+color;
+
+            Segments.push(<PieSegment angle={percentages[i]} offset={offset} color={color}></PieSegment>);
```

Comment:
If there's no plans to put anything inside a set of `<PieSegment></PieSegment>` tags, one could just use `<PieSegment />`. If there is however, just ignore me. :laughing:

Reply:
Changed to a self closing tag